frontend/src/state/index.ts

```diff
@@ -176,66 +137,87 @@ export const getTakenCredits = (state: AppState): number =>
   );
-
-/**
- * Get the current schedule from the Appstate
- * @param state the AppState
- */
-export const getScheduleDataFromState = (state: AppState): ScheduleSlice => {
-  return state.schedule.present;
 };
 
 /**
  * Get the current schedule's major from the AppState
  * @param state the AppState
  */
-export const getScheduleMajorFromState = (state: AppState): string => {
-  return getScheduleDataFromState(state).major;
+export const getActivePlanMajorFromState = (state: AppState) => {
+  return getActivePlanFromState(state).major;
 };
 
 /**
  * Get the current schedule's coop cycle from the AppState
  * @param state the AppState
  */
-export const getScheduleCoopCycleFromState = (state: AppState): string => {
-  return getScheduleDataFromState(state).coopCycle;
+export const getActivePlanCoopCycleFromState = (state: AppState) => {
+  return getActivePlanFromState(state).coopCycle;
 };
 
 /**
  * Get the list of schedule names from the AppState
  * @param state the AppState
  */
-export const getSchedulesFromState = (state: AppState): NamedSchedule[] =>
-  Object.values(state.schedules.schedules);
+export const getUserPlansFromState = (state: AppState): IPlanData[] =>
+  Object.values(state.userPlansState.plans);
 
 /**
  * Get the active schedule from the AppState
  * @param state the AppState
  */
-export const getActiveScheduleFromState = (state: AppState): NamedSchedule => {
-  return state.schedules.schedules[state.schedules.activeSchedule!];
+export const getActivePlanFromState = (state: AppState): IPlanData => {
+  return state.userPlansState.plans[state.userPlansState.activePlan!];
 };
 
-export const getClosedYearsFromState = (state: AppState): Set<number> => {
-  return new Set(state.schedule.present.closedYears);
+export const getActivePlanScheduleFromState = (state: AppState) => {
+  return getActivePlanFromState(state).schedule;
+};
+
+export const getClosedYearsFromState = (state: AppState) => {
+  return new Set(
+    state.userPlansState.closedYears[state.userPlansState.activePlan!]
+  );
 };
 
 /**
  * Get the active current class counter from the AppState
  * @param state the AppState
  */
-export const getCurrentClassCounterFromState = (state: AppState): number => {
-  return state.schedule.present.currentClassCounter;
+export const getCurrentClassCounterFromState = (state: AppState) => {
+  return getActivePlanFromState(state).courseCounter;
 };
 
 /**
  * Get whether the current user is an advisor or not
  * @param state the AppState
  */
-export const getIsAdvisorFromState = (state: AppState): boolean | undefined => {
-  return state.user.isAdvisor;
+export const getIsAdvisorFromState = (state: AppState) => {
+  return getUserFromState(state).isAdvisor;
 };
+
+/*
+ * SAFE GET FUNCTIONS
+ * Should only need to use these where there is not guaranteed to have an active plan (like the home screen)
+ * TODO: look into if a separate component (prior to getting to the HomeComponent) to load plan data would be worth it
```

Comment:
What if we just loaded all plan info on the loading screen on the redirect page. Once we get the user, we can get all necessary plan info/etc... We can do this in a separate PR but I think this would simplify the flow.

Reply:
I did consider this when coming up with solutions, it seemed like a lot of changes at the time. I do think that your idea is the best though, basically for the reason that you shouldn't be able to go to the HomeScreen without an active plan (which logically makes sense).

Reply:
I added a card for this in Trello